### PR TITLE
docs: edit copy and example in hierarchical injectors guide

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Currently, there's a provider missing in the example and @IgorMinar requested some copy edits.

Issue Number: #32461

## What is the new behavior?
Adds a missing provider in the example and edits copy.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/hierarchical-dependency-injection.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Hierarchical injectors
 
 Injectors in Angular have rules that you can leverage to
-achieve the desired visibility in your apps.
+achieve the desired visibility of provisioned objects in your apps.
```
Comment (petebacondarwin) [inline aio/content/guide/hierarchical-dependency-injection.md]: is "provisioned objects" a thing?
I would have gone with "provided objects" or "injectables".

**Review by petebacondarwin — APPROVED** [review]
One NIT.

File: aio/content/guide/hierarchical-dependency-injection.md [hunk 2]
```diff
@@ -1007,15 +1011,15 @@ which is in the `providers` array, the result was `null` because
...
-<app-root @NgModule(MyAppModule)
+<app-root @NgModule(AppModule)
         @Inject(AnimalService=>"🐳")>
-  <#VIEW>
-    <!-- ^^@Host()+@SkipSelf() stop here^^ -->
+  <#VIEW @Provide(AnimalService="🦔")>
+    <!-- ^^@SkipSelf() starts here,  @Host() stops here^^ -->
     <app-child>
       <#VIEW @Provide(AnimalService="🐶")
              @Inject(AnimalService, @SkipSelf, @Host, @Optional)=>"🐶">
```
Comment (ukyo) [inline aio/content/guide/hierarchical-dependency-injection.md]: Maybe `@Inject(AnimalService, @SkipSelf, @Host, @Optional)=>"🦔"`

Comment (kapunahelewong) [reply aio/content/guide/hierarchical-dependency-injection.md]: You're right! Added!

File: aio/content/guide/hierarchical-dependency-injection.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/hierarchical-dependency-injection.md]: Changed to "injectables". 😊

**Comment by kapunahelewong** [comment]
Caretaker, this is a docs only change.

**Comment by kapunahelewong** [comment]
Caretaker, this is a docs only change. Thank you!

**Pull request closed without merging**